virtool/references/db.py

```diff
@@ -433,6 +435,83 @@ async def update_reference(self):
         })
 
 
+class CreateIndexJSONTask(virtool.tasks.task.Task):
+
+    def __init__(self, app, task_id):
+        super().__init__(app, task_id)
+
+        self.steps = [
+            self.check_index_json_files,
+            self.create_index_json_files
+        ]
+
+        self.index_without_json = []
+
+    async def check_index_json_files(self):
+        db = self.db
+        settings = self.app["settings"]
+
+        async for index in db.indexes.find():
+            index_id = index["_id"]
+            ref_id = index["reference"]["id"]
+            index_path = os.path.join(settings["data_path"], "references", ref_id, index_id)
+            has_json = True
```

Comment:
I think it is safer `has_json` to only be set at line 508.

If this task as aborted and  the `json.gz` file is only partially written, the next time it runs the partial file will be assumed to be a complete file just because it exists at the expected path.

I think it is safer to check for `has_json` in the DB and overwrite and existing file if `has_json` is not defined or `False`. You will also save time not checking against the filesystem for every index.

What do you think?

Reply:
It makes sense! I didn't think of that point. I will fix it. Thanks!